File: mobile/lib/models/place.dart
class Place {
  final String id;
  final String name;
  final String category;
  final String? subcategory;
  final String? description;
  final double lat;
  final double lng;
  final String? address;
  final List<String>? images;
  final String? source;

  Place({
    required this.id,
    required this.name,
    required this.category,
    this.subcategory,
    this.description,
    required this.lat,
    required this.lng,
    this.address,
    this.images,
    this.source,
  });

  factory Place.fromJson(Map<String, dynamic> json) {
    return Place(
      id: json['id'] as String,
      name: json['name'] as String,
      category: json['category'] as String,
      subcategory: json['subcategory'] as String?,
      description: json['description'] as String?,
      lat: (json['lat'] as num).toDouble(),
      lng: (json['lng'] as num).toDouble(),
      address: json['address'] as String?,
      images: (json['images'] as List?)?.map((e) => e as String).toList(),
      source: json['source'] as String?,
    );
  }
}

File: mobile/src/types/place.ts
/** Shape returned by the `nearby_places` and `search_places` RPCs. */
export interface Place {
  id: string;
  name: string;
  category: string;
  subcategory: string | null;
  description: string | null;
  lat: number;
  lng: number;
  address: string | null;
  images: string[] | null;
  source: string | null;
  created_by?: string | null;
  created_at?: string | null;
}

/** A place plus its cosine similarity, from GET /recommendations. */
export interface SimilarPlace extends Omit<Place, 'id' | 'source'> {
  place_id: string;
  similarity_score: number;
}

export type ReportStatus = 'reported' | 'in_progress' | 'fixed';

export interface ProblemReport {
  id: string;
  category: string | null;
  description: string | null;
  photo_url: string | null;
  status: ReportStatus;
  created_at: string;
  resolved_at: string | null;
  lat: number | null;
  lng: number | null;
}
